### PR TITLE
Added network projectId to gcp and ocean providers

### PR DESCRIPTION
**Pull request by ebarriosjr**

### CODE REVIEW

File: service/ocean/providers/gcp/ocean.go [hunk 1]
```diff
@@ -970,6 +971,13 @@ func (o *NetworkInterface) SetNetwork(v *string) *NetworkInterface {
 	return o
 }
 
+func (o *NetworkInterface) SetProjectID(v *string) *NetworkInterface {
+	if o.ProjectId = v; o.ProjectId == nil {
+		o.nullFields = append(o.nullFields, "subnetwork_project")
```
Comment (liranp) [inline service/ocean/providers/gcp/ocean.go:976]:
```suggestion
		o.nullFields = append(o.nullFields, "ProjectID")
```

File: service/ocean/providers/gcp/ocean.go [hunk 2]
```diff
@@ -970,6 +971,13 @@ func (o *NetworkInterface) SetNetwork(v *string) *NetworkInterface {
 	return o
 }
 
+func (o *NetworkInterface) SetProjectID(v *string) *NetworkInterface {
+	if o.ProjectId = v; o.ProjectId == nil {
```
Comment (liranp) [inline service/ocean/providers/gcp/ocean.go:975]:
```suggestion
	if o.ProjectID = v; o.ProjectID == nil {
```

File: service/ocean/providers/gcp/ocean.go [hunk 3]
```diff
@@ -970,6 +971,13 @@ func (o *NetworkInterface) SetNetwork(v *string) *NetworkInterface {
 	return o
 }
 
+func (o *NetworkInterface) SetProjectID(v *string) *NetworkInterface {
```
Comment (liranp) [inline service/ocean/providers/gcp/ocean.go:974]:
```suggestion
func (o *NetworkInterface) SetProjectId(v *string) *NetworkInterface {
```

File: service/ocean/providers/gcp/ocean.go [hunk 4]
```diff
@@ -196,6 +196,7 @@ type NetworkInterface struct {
 	AccessConfigs []*AccessConfig `json:"accessConfigs,omitempty"`
 	AliasIPRanges []*AliasIPRange `json:"aliasIpRanges,omitempty"`
 	Network       *string         `json:"network,omitempty"`
+	ProjectId     *string         `json:"projectId,omitempty"`
```
Comment (liranp) [inline service/ocean/providers/gcp/ocean.go:199]:
```suggestion
	ProjectID     *string         `json:"projectId,omitempty"`
```

File: service/elastigroup/providers/gcp/gcp.go [hunk 5]
```diff
@@ -1552,6 +1553,14 @@ func (o *NetworkInterface) SetNetwork(v *string) *NetworkInterface {
 	return o
 }
 
+// SetProjectID sets the projectID fo the network
+func (o *NetworkInterface) SetProjectID(v *string) *NetworkInterface {
+	if o.ProjectId = v; o.ProjectId == nil {
```
Comment (liranp) [inline service/elastigroup/providers/gcp/gcp.go:1558]:
```suggestion
	if o.ProjectID = v; o.ProjectID == nil {
```

File: service/elastigroup/providers/gcp/gcp.go [hunk 6]
```diff
@@ -1552,6 +1553,14 @@ func (o *NetworkInterface) SetNetwork(v *string) *NetworkInterface {
 	return o
 }
 
+// SetProjectID sets the projectID fo the network
+func (o *NetworkInterface) SetProjectID(v *string) *NetworkInterface {
```
Comment (liranp) [inline service/elastigroup/providers/gcp/gcp.go:1557]:
```suggestion
func (o *NetworkInterface) SetProjectId(v *string) *NetworkInterface {
```

File: service/elastigroup/providers/gcp/gcp.go [hunk 7]
```diff
@@ -1552,6 +1553,14 @@ func (o *NetworkInterface) SetNetwork(v *string) *NetworkInterface {
 	return o
 }
 
+// SetProjectID sets the projectID fo the network
```
Comment (liranp) [inline service/elastigroup/providers/gcp/gcp.go:1556]:
```suggestion
// SetProjectId sets the project identifier of the network.
```

File: service/elastigroup/providers/gcp/gcp.go [hunk 8]
```diff
@@ -266,6 +266,7 @@ type NetworkInterface struct {
 	AccessConfigs []*AccessConfig `json:"accessConfigs,omitempty"`
 	AliasIPRanges []*AliasIPRange `json:"aliasIpRanges,omitempty"`
 	Network       *string         `json:"network,omitempty"`
+	ProjectId     *string         `json:"projectId,omitempty"`
```
Comment (liranp) [inline service/elastigroup/providers/gcp/gcp.go:269]:
```suggestion
	ProjectID     *string         `json:"projectId,omitempty"`
```

File: service/elastigroup/providers/gcp/gcp.go [hunk 9]
```diff
@@ -1552,6 +1553,14 @@ func (o *NetworkInterface) SetNetwork(v *string) *NetworkInterface {
 	return o
 }
 
+// SetProjectID sets the projectID fo the network
+func (o *NetworkInterface) SetProjectID(v *string) *NetworkInterface {
+	if o.ProjectId = v; o.ProjectId == nil {
+		o.nullFields = append(o.nullFields, "subnetwork_project")
```
Comment (liranp) [inline service/elastigroup/providers/gcp/gcp.go:1559]:
```suggestion
		o.nullFields = append(o.nullFields, "ProjectID")
```

**Comment by ebarriosjr** [comment]
@liranp thanks for taking a look at the code. :)

**Comment by liranp** [comment]
Thank you, @ebarriosjr. I've published a new release, [v1.46.0](https://github.com/spotinst/spotinst-sdk-go/releases/tag/v1.46.0).

**Pull request merged**